src/Ledger/Utxo/Properties.lagda

```diff
@@ -219,63 +245,58 @@ module DepositHelpers
+    cbalance utxoInsᶜ + (cbalance (outs tx) + txfee tx + (tot + don))  ≡⟨ cong (λ x → cbalance utxoInsᶜ + (cbalance (outs tx) + txfee tx + x)) (+-comm tot don) ⟩
+    cbalance utxoInsᶜ + (cbalance (outs tx) + txfee tx + (don + tot))  ≡˘⟨ +-assoc (cbalance utxoInsᶜ) (cbalance (outs tx) + txfee tx) (don + tot) ⟩
+    cbalance utxoInsᶜ + (cbalance (outs tx) + txfee tx) + (don + tot)  ≡˘⟨ cong (_+ (don + tot)) (+-assoc (cbalance utxoInsᶜ) (cbalance (outs tx)) (txfee tx)) ⟩
+    cbalance utxoInsᶜ + cbalance (outs tx) + txfee tx + (don + tot)    ≡˘⟨ cong (λ x → x + (txfee tx) + (don + tot)) (balance-∪ {utxoInsᶜ} {outs tx} h) ⟩
+    cbalance (utxoInsᶜ ∪ᵐˡ outs tx) + txfee tx + (don + tot)           ≡˘⟨ +-assoc (cbalance (utxoInsᶜ ∪ᵐˡ outs tx) + txfee tx) don tot ⟩
+    cbalance (utxoInsᶜ ∪ᵐˡ outs tx) + txfee tx + don + tot             ∎
+    where
+      γ : cbalance (utxoInsᶜ ∪ᵐˡ utxoIns) + ref ≡ cbalance utxo + ref
+      γ = cong (_+ ref) (balance-cong-coin {utxo = utxoInsᶜ ∪ᵐˡ utxoIns} {utxo' = utxo}
           let open IsEquivalence ≡ᵉ-isEquivalence renaming (trans to _≡ᵉ-∘_)
-          in (disjoint-∪ᵐˡ-∪ (disjoint-sym res-ex-disjoint) ≡ᵉ-∘ ∪-sym) ≡ᵉ-∘ res-ex-∪ (_∈? txins tx)) ⟩
-    cbalance ((utxo ∣ txins tx ᶜ) ∪ᵐˡ (utxo ∣ txins tx)) + ref
-      ≡⟨ cong (_+ ref) (balance-∪ {utxo ∣ txins tx ᶜ} {utxo ∣ txins tx} (flip res-ex-disjoint)) ⟩
-    cbalance (utxo ∣ txins tx ᶜ) + cbalance (utxo ∣ txins tx) + ref ≡t⟨⟩
-    cbalance (utxo ∣ txins tx ᶜ) + (cbalance (utxo ∣ txins tx) + ref)
-      ≡⟨ cong (cbalance (utxo ∣ txins tx ᶜ) +_)
-         (balValueToCoin {tx} {utxoSt} {UTxOEnv.pparams Γ} noMintAda newBal) ⟩
-    cbalance (utxo ∣ txins tx ᶜ) + (cbalance (outs tx) + txfee tx + tot + txdonation tx) ≡t⟨⟩
-    (cbalance (utxo ∣ txins tx ᶜ) + cbalance (outs tx) + txfee tx) + tot + txdonation tx
-      ≡˘⟨ cong (λ x → (x + txfee tx) + tot + txdonation tx) (balance-∪ {utxo ∣ txins tx ᶜ} {outs tx} h) ⟩
-    (cbalance ((utxo ∣ txins tx ᶜ) ∪ᵐˡ outs tx) + txfee tx) + tot + txdonation tx ≡t⟨⟩
-    (cbalance ((utxo ∣ txins tx ᶜ) ∪ᵐˡ outs tx) + txfee tx) + (tot + txdonation tx)
-      ≡⟨ cong ((cbalance ((utxo ∣ txins tx ᶜ) ∪ᵐˡ outs tx) + txfee tx) +_) (+-comm tot (txdonation tx)) ⟩
-    (cbalance ((utxo ∣ txins tx ᶜ) ∪ᵐˡ outs tx) + txfee tx) + (txdonation tx + tot) ≡t⟨⟩
-    (cbalance ((utxo ∣ txins tx ᶜ) ∪ᵐˡ outs tx) + txfee tx) + txdonation tx + tot ∎
-
-  rearrange0 : txfee tx + txdonation tx + (tot + remDepTot) + fees ≡
-        fees + txfee tx + getCoin deposits' + txdonation tx
+          in (disjoint-∪ᵐˡ-∪ (disjoint-sym res-ex-disjoint) ≡ᵉ-∘ ∪-sym) ≡ᵉ-∘ res-ex-∪ (_∈? txins tx))
+
+
+  rearrange0 : txfee tx + don + (tot + remDepTot) + fees ≡ fees + txfee tx + getCoin deposits' + don
   rearrange0 = begin
-    txfee tx + txdonation tx + (tot + remDepTot) + fees
-      ≡⟨ +-comm (txfee tx + txdonation tx + (tot + remDepTot)) fees ⟩
-    fees + (txfee tx + txdonation tx + (tot + remDepTot)) ≡t⟨⟩
```

Comment:
`≡t⟨⟩` is another tactic I couldn't get working with the overloaded `_+_` (cf. [the remark above](https://github.com/input-output-hk/formal-ledger-specifications/pull/173/files#r1290266719))

Reply:
reverted ∵ change requests